pros/src/async_runtime/executor.rs

```diff
@@ -0,0 +1,87 @@
+use core::{
+    cell::RefCell,
+    future::Future,
+    pin::Pin,
+    sync::atomic::{AtomicBool, Ordering},
+    task::{Context, Poll},
+};
+
+use alloc::{collections::VecDeque, sync::Arc};
+use async_task::{Runnable, Task};
+use waker_fn::waker_fn;
+
+use crate::os_task_local;
+
+use super::reactor::Reactor;
+
+os_task_local! {
+    pub(crate) static EXECUTOR: Executor = Executor::new();
+}
+
+pub(crate) struct Executor {
+    queue: RefCell<VecDeque<Runnable>>,
+    pub(crate) reactor: RefCell<Reactor>,
+}
+
+impl !Send for Executor {}
+impl !Sync for Executor {}
+
+impl Executor {
+    pub fn new() -> Self {
+        Self {
+            queue: RefCell::new(VecDeque::new()),
+            reactor: RefCell::new(Reactor::new()),
+        }
+    }
+
+    pub fn spawn<T>(&'static self, future: impl Future<Output = T> + 'static) -> Task<T> {
+        // SAFETY: `runnable` will never be moved off this thread or shared with another thread because of the `!Send + !Sync` bounds on `Self`.
+        //         Both `future` and `schedule` are `'static` so they cannot be used after being freed.
+        //   TODO: Make sure that the waker can never be sent off the thread.
```

Comment:
I dont know if this needs more work but might be good to fix after merge

Reply:
This will be fixed after this is merged and I split sync and async functionality from each other. I will put them into separate features that are mutually exclusive so that this cant really ever be an issue.